src/contracts/ops/DelegateDeployer.sol

```diff
@@ -0,0 +1,31 @@
+pragma solidity 0.6.2;
+pragma experimental ABIEncoderV2;
+
+// A delegate deployer that reverts upon failure
+// and emits the address if successful.
+// @author Patrick McCorry
+contract DelegateDeployer {
```

Comment:
we already have a deployer in the codebase. (e.g. deployerAddress). So would be good to avoid that collision

Reply:
You should look for the difference in behaviour rather than usage, I think the only difference between this and the other deployer is that this one reverts. So I would just name it that to avoid confusion: RevertDeployer.

Reply:
The other difference is the value field - which only makes sense to use if your doing delegatecall. otherwise you could run into a mistmatch between msg.value / value. so its designed to only be used by delegatecall safely